### ISSUE TITLE
NVDA should highlight or inverse current html object and automatically set cursor to current position for faster navigation

### ISSUE THREAD

**Issue by nvdaTrac** [nvaccess/test_nvdaTracImport#261]
**Reported by daTscharlie on 21 Sep 2008 19:02 UTC**
None

**Comment by nvdaTrac** [comment]
**Comment by jteh on 21 Sep 2008 20:52 UTC**
In Firefox, NVDA already moves the focus to the current HTML object if it is focusable and it also scrolls the window to the current object. NVDA can't highlight an HTML object except for setting focus to it, which is only possible if it is focusable. You have not specified what fails to work for you, nor what you mean by moving the cursor, as there is no real cursor in a web browser unless you mean the Firefox caret. You also haven't specified which browser you are referring to. Please be more specific.

**Comment by nvdaTrac** [comment]
**Comment by daTscharlie on 22 Sep 2008 08:17 UTC**
Replying to [jteh](comment:1):
> In Firefox, NVDA already moves the focus to the current HTML object if it is focusable and it also scrolls the window to the current object. NVDA can't highlight an HTML object except for setting focus to it, which is only possible if it is focusable. You have not specified what fails to work for you, nor what you mean by moving the cursor, as there is no real cursor in a web browser unless you mean the Firefox caret. You also haven't specified which browser you are referring to. Please be more specific.

Hello John,

here is my current environment:

Windows XP home SP3 with Windows High Contrast Black theme at 200 dpi resolution
NVIDIA graphics card (hardware zoom with KeyStone)
WebbIE 3.7.4
Firefox 3.0.1 with FireVox Bundle 4.0 Extension
Neospeech Kate and Paul 16 kHz
Scansoft Steffi 16 kHz
IBM ViaVoice 6.6.1.0 Multilingual
NVDA 0.6p2

Currently I'm trying out various combinations to obtain more comfort:

For example online releases of Dailies with WebbIE and NVDA

For a quick overview: WebbIE with it's function switch (ctrl+k) crop page view (only text - no links, since mostly all newspapers use over 300 links and are overwhelmed with advertising). In case I'll need more informations I shurely switch back with ctrl+k to uncrop page view and the cursor remains at the current HTML position, so I've just to move the cursor down to hear the links. Hereby is the current object inverted for visual support.

For example online releases of Dailies with Firefox, FireVox and NVDA

I use a big crosshair mouse cursor for better orientation, and FireVox since the current HTML objects (select current sentences) are inverted. I jump with the Virtual Buffer quick keys on the titles and then ctrl+shift+a (auto read in FireVox) to hear the text. Unfortunately, there are always commercials between the text blocks. To skip these advertising blocks now, I set the mouse cursor by hand to the final visual orientation point. FireVox auto read then continues reading at this point - unfortunately, sometimes I hit nothing at all and then I'll have to start over again. Liquid text aloud without commercial breaks will be less successful.
